dask/utils.py

```diff
@@ -625,3 +630,83 @@ def insert(tup, loc, val):
     L = list(tup)
     L[loc] = val
     return tuple(L)
+
+
+def infer_storage_options(urlpath, storage_options=None):
+    """ Infer storage options from URL path and merge it with existing storage
+    options.
+
+    Parameters
+    ----------
+    path: str or unicode
+        Either local absolute file path or URL (hdfs://namenode:8020/file.csv)
+    storage_options: dict (optional)
+        Its contents will get merged with the infered information from the
+        given path
+
+    Returns
+    -------
+    Storage options dict.
+
+    >>> so = infer_storage_options('/mnt/datasets/test.csv')
+    >>> so.pop('protocol')
+    'file'
+    >>> so.pop('path')
+    '/mnt/datasets/test.csv'
+    >>> so
+    {}
+    >>> infer_storage_options('./test.csv')['path']
+    './test.csv'
+    >>> infer_storage_options('../test.csv')['path']
+    '../test.csv'
+    >>> so = infer_storage_options(
+    ...          'hdfs://username:pwd@node:123/mnt/datasets/test.csv?q=1',
+    ...          storage_options={'extra': 'value'})
+    >>> so.pop('protocol')
+    'hdfs'
+    >>> so.pop('username')
+    'username'
+    >>> so.pop('password')
+    'pwd'
+    >>> so.pop('host')
+    'node'
+    >>> so.pop('port')
+    123
+    >>> so.pop('path')
+    '/mnt/datasets/test.csv'
+    >>> so.pop('url_query')
+    'q=1'
+    >>> so.pop('extra')
+    'value'
+    >>> so
+    {}
+    """
+    parsed_path = urlsplit(urlpath)
+
+    _storage_options = {
+        'protocol': parsed_path.scheme or 'file',
+        'path': parsed_path.path,
+    }
```

Comment:
I hate one-letter names which don't make sense until you read the whole function (instead of reading just one line). (The same story for me are shortcuts for `numpy` -> `np` and `pandas` -> `pd`, but I bear with it if the code is already written)